CMakeLists.txt

```diff
@@ -44,27 +43,31 @@ endif()
 if(HAIKU)
     # We add this explicitly, since we don't want CMake to think this
     #  is a C++ project unless we're on Haiku.
-    set(PHYSFS_CPP_SRCS src/physfs_platform_haiku.cpp)
+    list(APPEND PHYSFS_CPP_SRCS src/physfs_platform_haiku.cpp)
     find_library(BE_LIBRARY be)
     find_library(ROOT_LIBRARY root)
-    set(OPTIONAL_LIBRARY_LIBS ${OPTIONAL_LIBRARY_LIBS} ${BE_LIBRARY} ${ROOT_LIBRARY})
+    list(APPEND OPTIONAL_LIBRARY_LIBS ${BE_LIBRARY} ${ROOT_LIBRARY})
 endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL "WindowsPhone" OR CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
     set(WINRT TRUE)
 endif()
 
 if(WINRT)
-    set(PHYSFS_CPP_SRCS src/physfs_platform_winrt.cpp)
+    list(APPEND PHYSFS_CPP_SRCS src/physfs_platform_winrt.cpp)
 endif()
 
-if(UNIX AND NOT WINDOWS AND NOT APPLE)  # (MingW and such might be UNIX _and_ WINDOWS!)
+if(UNIX AND NOT MSVC AND NOT APPLE)  # (MingW and such might be UNIX _and_ WINDOWS!)
```

Comment:
We don't want pthreads on mingw, so changing this check to MSVC is probably not correct.

Reply:
I run physfs with mingw and neither variants causes the if to be true. and it works. might be my environment though.

Reply:
To get this pr going, do you want me to add [`AND NOT MINGW`](https://cmake.org/cmake/help/latest/variable/MINGW.html) to the `if`?

Reply:
Sorry, I missed these replies before; yes, let's add that for clarity even if it isn't strictly necessary.

Reply:
```suggestion
if(UNIX AND NOT MSVC AND NOT MINGW AND NOT APPLE)  # (MingW and such might be UNIX _and_ WINDOWS!)
```

Reply:
> To get this pr going, do you want me to add [`AND NOT MINGW`](https://cmake.org/cmake/help/latest/variable/MINGW.html) to the `if`?

Why not simply change it not `NOT WIN32` ?

Reply:
What is cygwin? Is that WIN32 and UNIX?
cygwin has a [pthread package](https://www.cygwin.com/packages/summary/mingw64-x86_64-pthreads.html).

Reply:
> What is cygwin? Is that WIN32 and UNIX? cygwin has a [pthreads package](https://www.cygwin.com/packages/summary/mingw64-x86_64-pthreads.html).

Never messed with cygwin. Maybe cmake's platform module would help with that:
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/CYGWIN.cmake

Reply:
CYGWIN sets WIN32 if you define `__USE_CMAKE_LEGACY_CYGWIN_WIN32` and `CMAKE_LEGACY_CYGWIN_WIN32`.
In that case, you're probably trying to build an older project.

So replacing with `NOT WIN32` is fine and `cygwin` can happily use its pthread library.

Reply:
OK.  (Don't know whether @icculus supports cygwin as cygwin, though.)

Reply:
This is fine with me, let's do `NOT WIN32` and get this PR merged.